### ISSUE TITLE
Problems assigning variables

### ISSUE THREAD

**Issue by baubakg** [jenkinsci/lockable-resources-plugin#205]
Hi,

I have a pipeline with the following structure:

```
node {

        stage ("Build & Launch Tests") {

            def tests = [:]
            def nrOfThreads = 2

                for (int i = 1; i <= nrOfThreads; i++) {                    tests[i] = {
            
                        lock(label: 'afthost', variable: 'LOCKED_SERVER', quantity:1) {
            
                            lock(label: 'slave_node', variable: 'LOCKED_SLAVE', quantity:1) {
                                            
                                stage ("Build Tests "  + testGroup) {
                                    cleanWs()

                                    echo env.LOCKED_SLAVE
                                    echo env.LOCKED_SERVER

                                    if (env.LOCKED_SERVER == null) {
                                        error("Server not locked properly")
                                    } 

                                    if (env.LOCKED_SLAVE == null) {
                                        error("Slave not locked properly")
                                    }                           
                                
                                }
                                       
                            }
                        }
                    }
                }
            tests.failFast = true
            parallel tests                        
        }
   
}
```
At one time it caused the issue:

```
Trying to acquire lock on [Label: afthost, Quantity: 1]
Found 0 available resource(s). Waiting for correct amount: 1.
[Label: afthost, Quantity: 1] is locked, waiting...
Lock acquired on [Label: afthost, Quantity: 1]
[Pipeline] {
[Pipeline] lock
Trying to acquire lock on [Label: slave_node, Quantity: 1]
Found 0 available resource(s). Waiting for correct amount: 1.
[Label: slave_node, Quantity: 1] is locked, waiting...
Lock acquired on [Label: slave_node, Quantity: 1]
[Pipeline] {
[Pipeline] node
Running on Jenkins in /apps/jenkins/latest/jobs/TAAS_PARALLEL_CAMPAIGN_POC/workspace@2
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Build Tests PX_POOL1)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] echo
null
[Pipeline] echo
null
[Pipeline] dir
```

In the above log I print both locked resources, but they are null.

It has happened once, but I could not find any references to a similar problem. I have tried reproducing this in a smaller project but so far to no avail.

In summary I have:
- two group of locked resources
- since job is in parallel they are locked in parallel
- I use scripted pipeline

I just wanted to signal this, in case anyone knows about it.

**Comment by TobiX** [comment]
This sounds a lot like the bug fixed in [2.7](https://github.com/jenkinsci/lockable-resources-plugin/releases/tag/lockable-resources-2.7), very strange...

**Comment by baubakg** [comment]
Ah thanks. That explains a few things:
1. The issues arrived in 2.6
2. I Tried reproducing in 2.7 (hence my inability to reproduce) :)

Thanks I'll try and upgrade my plugin.

**Comment by TobiX** [comment]
For reference: The bug fixed in 2.7 has existed since at least 2.2 (maybe even since this plugin started supporting pipeline)

**Comment by baubakg** [comment]
Sorry for my late response. It worked so good I forgot to close this issue. Thanks. It works like a charm.
